tensorbay/opendataset/VGGFace2/loader.py

```diff
@@ -0,0 +1,196 @@
+    with open(os.path.join(root_path, "label", "identity_meta.csv"), encoding="utf-8") as fp:
+        lines = islice(csv.reader(fp), 1, None)
+        for line in lines:
+            # The normal format of each line of the file is
+            # '<class_id>,"<class_name>",<sample_num>,<flag>,<gender>,\n'
+            # but now there is an error type
+            # '<class_id>,"<class_,name>",<sample_num>,<flag>,<gender>\n'
+            if line[-1] != "":
+                # join the splitted "class_ and name"
+                line[1] = "".join(islice(line, 1, 3))
+                # Starting from the 4th element, each element moves forward one step
+                line.pop(2)
+                line[4] = line[4].rstrip("\n")
+            category_id = line[0]
+            category_name = line[1].strip('"')
+            attributes = dict(
+                zip(
+                    attribute_names,
+                    (
+                        int(line[2]),
+                        bool(int(line[3])),
+                        line[4],
+                    ),
+                )
+            )
+            all_classifications_id_to_name[category_id] = category_name
+            all_classifications[category_name] = Classification(
+                category=category_name,
+                attributes=attributes,
+            )
+    return all_classifications, all_classifications_id_to_name
+
+
+def _get_box2ds(root_path: str) -> Dict[str, LabeledBox2D]:
+    all_box2d_attributes: DefaultDict[str, AttributeType] = defaultdict(dict)
+    box2d_files = glob(os.path.join(root_path, "label", "attributes", "*.txt"))
+    for file in box2d_files:
+        attribute_name = os.path.basename(file).rstrip(".txt").split("-", 1)[1]
+        with open(file, encoding="utf-8") as fp:
+            for line in fp:
+                name_id, attribute_value = line.rstrip("\n").split("\t", 1)
+                all_box2d_attributes[name_id][attribute_name] = bool(int(attribute_value))
+    all_boxes = {}
+    for file in (
+        os.path.join(root_path, "label", "loose_bb_test.csv"),
+        os.path.join(root_path, "label", "loose_bb_train.csv"),
+    ):
+        with open(file, encoding="utf-8") as fp:
+            for items in islice(csv.reader(fp), 1, None):
+                name_id = items[0].strip('"')
+                coordinate = (float(item) for item in islice(items, 1, None))
+                box = LabeledBox2D.from_xywh(*coordinate)
+                box2d_attribute = all_box2d_attributes.get(f"{name_id}.jpg")
+                if box2d_attribute:
+                    box.attributes = box2d_attribute
+                all_boxes[f"{name_id}.jpg"] = box
+
+    return all_boxes
+
+
+def _get_keypoint2ds(root_path: str) -> Dict[str, LabeledKeypoints2D]:
+    all_keypoint2ds = {}
+    for file in (
+        os.path.join(root_path, "label", "loose_landmark_test.csv"),
+        os.path.join(root_path, "label", "loose_landmark_train.csv"),
```

Comment:
We join `root_path` and `"label"` many times, shall we set `os.path.join(root_path, "label")` as a variable?